build_tools/benchmarks/post_benchmark_comment_test.py

```diff
@@ -66,45 +66,60 @@ def test_get_previous_comment_on_pr(self):
             "login": "user"
         },
         "body": "comment id: 1234"
-    }, {
+    }]
+    second_response = mock.create_autospec(requests.Response)
+    second_response.status_code = http.client.OK
+    second_response.json.return_value = [{
         "id": 3,
         "user": {
             "login": "bot"
         },
         "body": "comment id: 1234"
     }]
+
+    def _handle_get(endpoint: str, payload: Any):
+      if payload["page"] == 1:
+        return first_response
+      if payload["page"] == 2:
+        return second_response
+      raise ValueError("Unexpected page")
+
     requester = mock.create_autospec(post_benchmark_comment.APIRequester)
-    requester.get.return_value = response
+    requester.get.side_effect = _handle_get
```

Comment:
FYI, you can give side_effect an iterable:

```suggestion
    requester.get.side_effect = [first_response, second_response]
```

It's a bit less robust because it doesn't actually ensure you're calling with the correct page numbers, but it's much less verbose. Writing good tests is hard...

Reply:
SG. Done.